feat/feature.go

```diff
@@ -140,33 +165,42 @@ func PositionWithin(f, ref Feature, position int) (pos int, ok bool) {
 	panic("feat: feature chain too long")
 }
 
-// BaseOrientationOf returns the orientation of the given feature relative to
-// the deepest orientable location and the reference feature, which may be
-// the feature itself if it is not an Orienter or has a nil Location.
-// The returned orientation will always be Forward or Reverse.
+// BaseOrientationOf returns the orientation of f relative to the shallowest
+// non orientable feature location and a reference feature which will be the
+// shallowest non orientable feature location if it is not nil or its parent
+// feature if it is nil. The reference feature can be used by different calls
+// to BaseOrientationOf to verify that returned orientations are comparable.
+// The returned orientation will be NotOriented if f is not orientable.
 // BaseOrientationOf will panic if the feature chain is deeper than 1000 links.
 func BaseOrientationOf(f Feature) (ori Orientation, ref Feature) {
+	o, ok := f.(Orienter)
+	if !ok || o.Orientation() == NotOriented {
+		return NotOriented, nil
```

Comment:
I'm not sure this is right behaviour. f is not orientable, but it's not because of the chain it's because of f. I think in this case it should walk down until the first orientable or last non-nil. Though prepared to be convinced otherwise.